### PR TITLE
https://unit.atlassian.net/browse/NP-45477

### PR DESCRIPTION
**Pull request by StigNorland**
publicationDate should accept a year or a shortDate (YYYY-MM-DD)
from cristin it should be just year, however to keep it consistent, it should accept a date, just like created and modified

"no query" should return all.

FUNDING_SOURCE syntax error

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java [hunk 1]
```diff
@@ -36,25 +38,23 @@ public enum ResourceParameterKey implements ParameterKey {
     INVALID(STRING),
     // Parameters converted to Lucene query
     CATEGORY(STRING, "entityDescription.reference.publicationInstance.type"),
-    CONTRIBUTOR(STRING, "entityDescription.contributors.identity.id"
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:39]: no more search by contributor id?

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java [hunk 2]
```diff
@@ -36,25 +38,23 @@ public enum ResourceParameterKey implements ParameterKey {
...
-                        + "|entityDescription.contributors.identity.name"),
+    CONTRIBUTOR(STRING, "entityDescription.contributors.identity.name"),
     CREATED_BEFORE(DATE, LESS_THAN, "createdDate"),
     CREATED_SINCE(DATE, GREATER_THAN_OR_EQUAL_TO, "createdDate"),
     DOI(STRING, "entityDescription.reference.doi"),
     FUNDING(STRING, "fundings.identifier"),
-    FUNDING_SOURCE(STRING, "fundings.source.identifier|fundings.source.labels.*"),
+    FUNDING_SOURCE(STRING, "fundings.source.identifier|fundings.source.labels"),
     ID(STRING, "identifier"),
-    INSTITUTION(STRING, "entityDescription.contributors.affiliation.id"
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:47]: no more search on id?

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java [hunk 3]
```diff
@@ -36,25 +38,23 @@ public enum ResourceParameterKey implements ParameterKey {
...
     // Parameters converted to Lucene query
     CATEGORY(STRING, "entityDescription.reference.publicationInstance.type"),
-    CONTRIBUTOR(STRING, "entityDescription.contributors.identity.id"
-                        + "|entityDescription.contributors.identity.name"),
+    CONTRIBUTOR(STRING, "entityDescription.contributors.identity.name"),
     CREATED_BEFORE(DATE, LESS_THAN, "createdDate"),
     CREATED_SINCE(DATE, GREATER_THAN_OR_EQUAL_TO, "createdDate"),
     DOI(STRING, "entityDescription.reference.doi"),
     FUNDING(STRING, "fundings.identifier"),
-    FUNDING_SOURCE(STRING, "fundings.source.identifier|fundings.source.labels.*"),
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:45]: syntax error fix

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java (hunk 1 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:39]: I need time to investigate this, it is really strange

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java (hunk 2 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:47]: I need time to investigate this, it is really strange (luciene syntax weirdness)

see examples here...

https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.identity.name:Muhammad%20Yahya)
https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.identity:Muhammad%20Yahya)
https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.identity.id:Muhammad%20Yahya)
[https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.[…]yDescription.contributors.identity.name:Muhammad%20Yahya)](https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.identity.id:Muhammad%20Yahya%20OR%20entityDescription.contributors.identity.name:Muhammad%20Yahya))
og denne blir helt vill
[https://api.dev.nva.aws.unit.no/search/resources/?query=((entityDescription.contributors[…]tyDescription.contributors.identity.id:Muhammad%20Yahya))](https://api.dev.nva.aws.unit.no/search/resources/?query=((entityDescription.contributors.identity.name:Muhammad%20Yahya)%20or%20(entityDescription.contributors.identity.id:Muhammad%20Yahya))) -> 25280
og denne enda mer!
[https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.[…]ityDescription.contributors.identity.id:Muhammad%20Yahya)](https://api.dev.nva.aws.unit.no/search/resources/?query=(entityDescription.contributors.identity.name:Muhammad%20Yahya)%20and%20(entityDescription.contributors.identity.id:Muhammad%20Yahya))  -> 79986

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 4]
```diff
@@ -87,6 +87,8 @@ private Stream<QueryBuilderWrapper> createQueryBuilderStream(ResourceAwsQuery qu
                 ? QueryBuilders
                     .multiMatchQuery(query.getValue(SEARCH_ALL).toString(), extractFields(field))
                     .operator(Operator.AND)
+                : query.noLucineParameter()
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:90]: hard to read this kind of conditional operator when it is nested

Comment (torarnet) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:90]: Dont really understand it, but if you know it works...

Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:90]: I agree...

**Review by torarnet — APPROVED** [review]

**Pull request merged**